src/main/java/com/mraof/minestuck/alchemy/GristHelper.java

```diff
@@ -135,58 +141,61 @@ public static void increase(Level level, PlayerIdentifier player, GristSet set)
 		data.setGristCache(newCache);
 	}
 	
-	public static void notify(MinecraftServer server, PlayerIdentifier player, GristSet set)
+	public static void notify(MinecraftServer server, PlayerIdentifier player, GristSet set, GristToast.EnumSource source, boolean increase)
 	{
 		if(MinestuckConfig.SERVER.showGristChanges.get())
 		{
-			Map<GristType, Long> reqs = set.getMap();
-			for(Entry<GristType, Long> pairs : reqs.entrySet())
+			
+			//don't know why this is here, it was copied over from when i merged notifyEditMode() and notify(), but im not sure what it does/did. Seems to be a sanity check.
+			SburbConnection sc;
+			EditData ed;
+			
+			if(source == GristToast.EnumSource.SERVER) {
+				sc = SkaianetHandler.get(server).getActiveConnection(player);
+				if(sc == null)
+					return;
+				
+				ed = ServerEditHandler.getData(server, sc);
```

Comment:
The edit data was previously used to send a grist notification to the edit player. It looks like you're now instead only sending grist notifications to the player which the grist cache belongs to.

Reply:
Okay, cool. I thought that was why it was there, but I wasn't sure. It's been removed.